endpoint.go

```diff
@@ -343,7 +343,10 @@ func (ep *endpoint) getNetworkFromStore() (*network, error) {
 		return nil, fmt.Errorf("invalid network object in endpoint %s", ep.Name())
 	}
 
-	return ep.network.ctrlr.getNetworkFromStore(ep.network.id)
+	ep.network.Lock()
```

Comment:
Can we reuse `network.getController()` ?
Ideally we could also do 
`ep.getNetwork().getController().getNetworkFromStore(ep.network.id)` 
but it looks too long of a concatenation...